bundles/org.openhab.binding.meater/src/main/java/org/openhab/binding/meater/internal/api/MeaterRestAPI.java

```diff
@@ -0,0 +1,204 @@
+        }
+        return false;
+    }
+
+    private void login() throws MeaterException {
+        try {
+            // Login
+            String json = "{ \"email\": \"" + configuration.email + "\",  \"password\": \"" + configuration.password
+                    + "\" }";
+            Request request = httpClient.newRequest(API_ENDPOINT + LOGIN).method(HttpMethod.POST);
+            request.header(HttpHeader.ACCEPT, JSON_CONTENT_TYPE);
+            request.header(HttpHeader.CONTENT_TYPE, JSON_CONTENT_TYPE);
+            request.content(new StringContentProvider(json), JSON_CONTENT_TYPE);
+
+            logger.trace("{}.", request.toString());
+
+            ContentResponse httpResponse = request.send();
+            if (!HttpStatus.isSuccess(httpResponse.getStatus())) {
+                throw new MeaterException("Failed to login " + httpResponse.getContentAsString());
+            }
+            // Fetch JWT
+            json = httpResponse.getContentAsString();
+            JsonObject jsonObject = JsonParser.parseString(json).getAsJsonObject();
+            JsonObject childObject = jsonObject.getAsJsonObject("data");
+            JsonElement tokenJson = childObject.get("token");
+            if (tokenJson != null) {
+                this.authToken = tokenJson.getAsString();
+            } else {
+                throw new MeaterException("Tken JSON is null");
+            }
+        } catch (TimeoutException | ExecutionException | JsonParseException e) {
+            throw new MeaterException(e);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new MeaterException(e);
+        }
+    }
+
+    private String getFromApi(String uri) throws MeaterException, MeaterAuthenticationException {
+        try {
+            for (int i = 0; i < MAX_RETRIES; i++) {
+                try {
+                    if ("".equals(authToken)) {
+                        login();
+                    }
+                    Request request = httpClient.newRequest(API_ENDPOINT + uri).method(HttpMethod.GET);
+                    request.header(HttpHeader.AUTHORIZATION, "Bearer " + authToken);
+                    request.header(HttpHeader.ACCEPT, JSON_CONTENT_TYPE);
+                    request.header(HttpHeader.CONTENT_TYPE, JSON_CONTENT_TYPE);
+                    request.header(HttpHeader.ACCEPT_LANGUAGE, localeProvider.getLocale().getLanguage());
+
+                    ContentResponse response = request.send();
+                    String content = response.getContentAsString();
+                    logger.trace("API response: {}", content);
+
+                    if (response.getStatus() == HttpStatus.UNAUTHORIZED_401) {
+                        // This will currently not happen because "WWW-Authenticate" header is missing; see below.
+                        logger.debug("getFromApi failed, HTTP status: 401");
+                        login();
```

Comment:
In case API will be fixed in the future, this should be adapted to same handling as in line 164..173:
```suggestion
                        throw new MeaterAuthenticationException("Authentication failed");
```

Reply:
Will make a mental note on that!

Reply:
Suggestion would be to fix this now so the same reauthentication mechanism is used as in the work-around for the RFC violation. This way nothing would be needed in the future. :-)